guides/Building Your Bot/CreatingACustomMemberGateway.md

```diff
@@ -0,0 +1,171 @@
+Something that is often asked or requested, is a custom member-based gateway; that is per member configs, so you would have different {@link Configuration} values for different guilds.
+
+This is simple to achieve with Klasa's advanced Gateway system. In this example, we will be make a per member gateway with a built in key for points. We will be remaking the {@tutorial CreatingPointsSystems} example, in this tutorial.
+
+> Note: I would not recommend using the default (JSON) provider for this, as you may end up reaching the limit of JSON files your OS can open.
+
+## Creating The Gateway
+
+Inside your entry point (Where you define `new Client()`, for example, `index.js`) add this:
+
+```javascript
+client.gateways.register('members', {
+	experience: {
+		type: 'Integer',
+		default: 10
+	},
+	level: {
+		type: 'Integer',
+		default: 1
+	}
+}, { provider: 'rethinkdb' });
+```
+
+And that's it, we now have a built in members gateway, however we have no way of accessing them via the [GuildMember](https://discord.js.org/#/docs/main/master/class/GuildMember) class. We'll tackle that next.
+
+## Creating Our Own GuildMember Class
+
+We have the Gateway, so we just need to use a {@link Extendable} to use it properly. Make a new Extendable in /extendables and call it `GuildMember`. Add this in:
+
+```javascript
+const { Extendable } = require('klasa');
+
+module.exports = class extends Extendable {
+
+	constructor(...args) {
+		super(...args, {
+            		appliesTo: ["GuildMember"],
+            		name: 'configs',
+            		enabled: true,
+         	  	klasa: false
+		});
+	}
+
+	get extend() {
+		return this.client.gateways.members.get(`${this.guild.id}-${this.id}`, true); //we pass "true" here to create an entry, if it doesn't exist.
+	}
+
+};
+
+```
+
+Alright! Now you have a fully functioning member-gateway. Now, we can create a per guild level system.
+
+## Level Up!
+
+We will be taking code from {@tutorial CreatingPointsSystems}, and modifying it a small bit to work with the newly created member-gateway.
+
+
+Create a new monitor, and add the follwing code:
+
+```javascript
+const { Monitor } = require('klasa');
+
+module.exports = class extends Monitor {
+
+	// Constructor
+
+	async run(message) {
+		// If the message was not sent in a TextChannel, ignore it.
+		if (!message.guild) return;
+
+		// Calculate the next value for experience.
+		const nextValue = message.member.configs.experience + 1;
+
+		// Cache the current level.
+		const currLevel = message.member.configs.level;
+
+		// Calculate the next level.
+		const nextLevel = Math.floor(0.1 * Math.sqrt(nextValue + 1));

```

Comment:
+1 again?

Reply:
Then fix it there too!

Reply:
Actually, I think this is part of the math behind the level up system, as it isn't actually modifying the configuration value `(nextValue + 1)` - not sure though. I'll leave this to kyra.